### PR TITLE
Feature/search tag

### PR DESCRIPTION
**Pull request by srsingh04**
This is the code for the UI of the tags. The tags will be rather personalized for each users depending on what activities they do.

### CODE REVIEW

**Comment by ntissieres** [comment]
I think we need to discuss more in details how we will implement the tag hierarchy, and follow it for the implementation of this feature. Like if there is more than a number of tags from the same category, we could group them and display the parent tag. When the user click on this parent tag, the children would appear. I will clarify it with you tomorrow if it's not clear.

**Review by ntissieres — CHANGES_REQUESTED** [review]
Maybe it's not a good idea to name the file Tags.kt, we already have a file named Tag to define the data class and moreover it's not truly indicative of what it encodes. MainscreenTagsFilter would be better for example

**Review by ntissieres — CHANGES_REQUESTED** [review]
Don't forget to remove everything that is commented. In order to merge the code into the main, the code should be clean

File: app/src/main/java/com/github/swent/echo/data/model/Tag.kt [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
 
-data class Tag(val tagId: String, val name: String)
+/**
+ * Represents a tag.
+ *
+ * @param tagId The ID of the tag.
+ * @param name The name of the tag.
+ * @param isSelected Whether the tag is selected.
+ */
+data class Tag(val tagId: String, val name: String, val isSelected: Boolean)
```
Comment (violoncelloCH) [inline app/src/main/java/com/github/swent/echo/data/model/Tag.kt:10]: I'm not sure if having `isSelected` on the data model makes sense. This is state only concerning the UI and nothing we want to store. See also my comment here: https://github.com/echo-swent/echo/pull/31#issuecomment-2047914993

**Review by ntissieres — CHANGES_REQUESTED** [review]
It's better to remove the padding on the top of the items, it will be done after when we add the search tag to the scaffold

**Review by ntissieres — APPROVED** [review]
Perfect !

**Pull request merged**